### ISSUE TITLE
vsscript_init failure

### ISSUE THREAD

**Issue by stax76** [vapoursynth/vapoursynth#614]
There is a staxrip user that is getting a vsscript_init failure in both portable and installed mode, he says that mpv.net does work, mpv uses lavf and not vfw to open vpy. Maybe you have an idea how to find the reason. As workaround, I added a VFW mode but since VFW mode is not the default and I don't know how many people are affected it would be good if the reason could be found. I don't expect it to be easy since it's not reproducible by a developer, so far it's reported by only one person.

https://forum.doom9.org/showthread.php?p=1918423#post1918423

https://github.com/staxrip/staxrip/issues/281

https://github.com/staxrip/staxrip/blob/master/FrameServer/VapourSynthServer.cpp

**Comment by myrsloik** [comment]
Missing visual studio runtimes? A difference in the working directory?

**Comment by stax76** [comment]
The last beta has all x64 binaries of a popular all in one VC++ redist package (abbodi1406/vcredist) included in a folder which is added per process as env var so Windows default DLL loading will find it. I'm waiting for the user who has reported the issue to try this beta. The alternative VFW mode is now also available.

**Comment by myrsloik** [comment]
I'd rather see you avoid the VFW mode with VapourSynth. It performs considerably worse in most cases and most exists for convenience...

**Comment by stax76** [comment]
I added VFW mode in case there are issues and for backward compatibility.

There are three options for avisynth and vapoursynth:

Use portable directly (default)
Use installed directly
Use installed via VFW

With vapoursynth there are no issues but with avisynth it's complicated because it's in system32 and also because ffmpeg blocks default DLL loading so PATH is not searched. So for avisynth portable mode I need to use soft links which need admin rights or developer mode. Only avs2pipemod supports to use a custom avisynth path via CLI, I will try to request it for all tools, this will be difficult, first request made today at nvenc.

staxrip does not do something performance critical like encoding, the preview and crop dialogs are fast enough even with VFW. There are some .NET tools like megui that have slow drawing code, one frame takes 200ms on a 4K screen, in staxrip it was modernized with Direct2D last year.

The user has found the problem fortunately, he has copied vsscript.dll to system32 in order to workaround a nvenc issue which was fixed meanwhile. I told him a better workaround would have been to modify PATH.

My frameserver.dll calls LoadLibrary("vsscript.dll") and the OS finds it because staxrip adds the folder per process to PATH, not sure if that is the best possible solution and if there are scenarios where it would fail.
